### DISCUSSION TITLE
Mission Combat Coalition Vs Separatists not loading/unlocking

### DISCUSSION THREAD

**Discussion by Myrik25** [FakeFishGames/Barotrauma#13620] [Code questions and modding support]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

The "Mission Combat Coalition Vs Separatists" not loading/unlocking when you have high faction with the Coalition and they want you to hunt the Separatists... it does work the other way around. 
![bug](https://github.com/FakeFishGames/Barotrauma/assets/43661529/30734634-2827-4a55-a57d-e4956ae29658)

### Reproduction steps

1. At an outpost I talk to the Bot who gives out the missions.
2. Once the mission to attack the aggressive Separatist sub is supposed to be given, it pops up the command panel and gives the error:  ‖color:#9370DB‖[Vanilla]‖color:end‖ Failed to unlock a mission with the tag "combatcoalitionvsseparatists": no matching missions found.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### -

During in Single or Multi-Player Campaign

### Version

v1.2.8.0 (Winter Update hotfix 2)

### -

During in Single or Multi-Player Campaign

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
‖color:#9370DB‖[Vanilla]‖color:end‖ Failed to unlock a mission with the tag "combatcoalitionvsseparatists": no matching missions found.
```

**Comment by Regalis11** [comment]
Thank you for the report!

I unfortunately could not find a way to reproduce this, the mission seems to be triggering with no issues.
![image](https://github.com/FakeFishGames/Barotrauma/assets/14351070/764a3ad4-5da8-4b64-ad09-9855102dc0be)

You mentioned you're using some kind of mods, what mods? I suspect some of them might be causing this, perhaps they override the combat missions and modify the tags.

**Comment by Myrik25** [comment]
> You mentioned you're using some kind of mods, what mods? I suspect some of them might be causing this, perhaps they override the combat missions and modify the tags.

Outside of downloading subs and assemblies from the workshop, the only mods I use that add/change gameplay I will list here in order:

Lua:
https://steamcommunity.com/sharedfiles/filedetails/?id=2559634234&searchtext=lua

MN Creatures:
https://steamcommunity.com/sharedfiles/filedetails/?id=2961385886&searchtext=Mn+creatures

Balanced Wreck Locator:
https://steamcommunity.com/sharedfiles/filedetails/?id=2941904990&searchtext=balanced+wreck

Beacons Extended:
https://steamcommunity.com/sharedfiles/filedetails/?id=2389600483&searchtext=beacons+extended

Shipwrecks Extended:
https://steamcommunity.com/sharedfiles/filedetails/?id=2095211492&searchtext=shipwrecks+extended

New Wrecks for Barotrauma:
https://steamcommunity.com/sharedfiles/filedetails/?id=2184257427

Real Sonar:
https://steamcommunity.com/sharedfiles/filedetails/?id=2936760984&searchtext=real+sonar

The MN Creatures is no longer updated by the creator, so it would be my first guess to remove but I'm not an expert at mods

**Comment by Myrik25** [comment]
I removed all mods from the current save and then used the console to "triggerevent missionevent_pirate1" to see if this might fix the issue but it still will not load the mission, it gives the same error message.

**Comment by SomeRandomNoobKekeke** [comment]
Wait, do you mean mission when you need to destroy enemy sub? I thought it's pirate1 and pirate1separatists
and you type unlockmission pirate1, unlockmission pirate1separatists on any outpost to unlock them

Or you meant some other mission?

**Comment by Myrik25** [comment]
Yes, to destroy the enemy faction sub.  I have 100 positive Coalition faction and hated by Separatists... The game pops up a console error from the pic above and fails to load the mission... so I was looking to see if there was a console command that would trigger it so I could test if removing the mods from the save fixes the issue or not.

**Comment by SomeRandomNoobKekeke** [comment]
Couldn't you just give us this save? Since i can't reproduce it either

**Comment by Myrik25** [comment]
> Couldn't you just give us this save? I can't reproduce it either

Sure
https://drive.google.com/file/d/1qsvYaz-of8hnLaYqe-uHIIZs-ZWxB2w5/view?usp=sharing

**Comment by SomeRandomNoobKekeke** [comment]
![mods](https://github.com/FakeFishGames/Barotrauma/assets/122838333/6127ae57-9ac6-4b95-a6ee-9584349115d5)
kek

Well, i opened it with no mods and outpost manager doesn't even try to give me pirate mission
Could it be that you started campaign with some mods, then disabled them but pirate missions remains? idk
I bet it caused by one of those mods, and i can't find them all by names. Unless you have collection with them it's gonna be very hard to find  and test them

and i can unlockmission pirate1, unlockmission pirate1separatists

also why do your bots have serversidelua affliction?
I think some lua mods can load even when disabled and don't pop up in campaign mod list, so you might want to go to main menu - luaCs settings - and enable "treat forced mods as normal" to make sure they are excluded

**Comment by Myrik25** [comment]
> ![mods](https://private-user-images.githubusercontent.com/122838333/317130101-6127ae57-9ac6-4b95-a6ee-9584349115d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE1NTA5MjEsIm5iZiI6MTcxMTU1MDYyMSwicGF0aCI6Ii8xMjI4MzgzMzMvMzE3MTMwMTAxLTYxMjdhZTU3LTlhYzYtNGI5NS1hNmVlLTk1ODQzNDkxMTVkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyN1QxNDQzNDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDFlODFlMDNkYjI2YzQ0ODVmMmM5MTQ3YmNmMDViNDFmMjI3NTI1NmUwMjg2N2Y3YWJjYjM1OGVkMmZhNjI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QUIxbdmYzEmP1ur-7btOjp5yghrPzW4DCcHH4hEVVz4) kek
> 
> Well, i opened it with no mods and outpost manager doesn't even try to give me pirate mission Could it be that you started campaign with some mods, then disabled them but pirate missions remains? idk I bet it caused by one of those mods, and i can't find them all by names. Unless you have collection with them it's gonna be very hard to find and test them
> 
> and i can unlockmission pirate1, unlockmission pirate1separatists
> 
> also why do your bots have serversidelua affliction? I think some lua mods can load even when disabled and don't pop up in campaign mod list, so you might want to go to main menu - luaCs settings - and enable "treat forced mods as normal" to make sure they are excluded

I posted the only mods that change gameplay I used above with links to them... the other things are only assemblies that don't affect gameplay, like  auto turrets, and other submarines from the workshop.  I already have the "treat forced mods as normal" enabled before posting.  

I don't know why it says the bots have serversidelua affliction... 
When I saw the error and how he tried to use the console to force the mission and it worked for him I tried removing the mods and do the same but with no luck.

I can put the mods back and reupload the save for you to test.  I only wanted to play the base vanilla mission of hunting the other faction's subs... I didn't add anything about pirates to this save, I think I had one about pirates like a year ago but I unsubbed to it.

**Reply by SomeRandomNoobKekeke** [reply]
> I can put the mods back and reupload the save for you to test

i don't think it will change the save

E.g. what is Piranha-Pirate?

**Comment by Myrik25** [comment]
> > I can put the mods back and reupload the save for you to test
> 
> i don't think it will change the save
> 
> E.g. what is Piranha-Pirate?

I don't know, that's the first I've heard of it.

What are the exact console commands to get the missions to hunt down the other factions?

**Reply by SomeRandomNoobKekeke** [reply]
try these unlockmission pirate1, unlockmission pirate1separatists

**Reply by Myrik25** [reply]
The first one does not work, but the pirate1separatists does.

The first one gives me the error:
[Piranha-Pirate] Error in mission "Pirate1". Character prefab "pirateengineerveteran" not found in NPC set "piratenpcs1".

It actually has the error for all prefabs with the "pirate......" list.

If I use the triggerevent missionevent_pirate1 command it gives me the error after I try to accept the mission:
[VANILLA] WARNING:  Failed to unlock a mission with tag "combatseparatistsvscoalition": no matching missions found.

I dunno what the Piranha-Pirate thing is from.

**Reply by SomeRandomNoobKekeke** [reply]
it could be some mod inside another mod
Or maybe it was added and then removed in some update in some mod

you know what? go to sub editor, and search submarine list for this "Piranha-Pirate"
if it's in the list then mod with it is still enabled

**Reply by Regalis11** [reply]
It seems that the mod "Piranha-Pirate" (which seems to be [a part of this collection](https://steamcommunity.com/sharedfiles/filedetails/?id=2560762670)) overrides the vanilla pirate missions, so this doesn't seem to be an issue with the vanilla game. I'll move the discussion to the modding section.

**Comment by Myrik25** [comment]
> try these unlockmission pirate1, unlockmission pirate1separatists



> try these unlockmission pirate1, unlockmission pirate1separatists



> It seems that the mod "Piranha-Pirate" (which seems to be [a part of this collection](https://steamcommunity.com/sharedfiles/filedetails/?id=2560762670)) overrides the vanilla pirate missions, so this doesn't seem to be an issue with the vanilla game. I'll move the discussion to the modding section.

I did not know I was subbed this mod... I've unsubbed from it and started testing the save without it.  It did not give me the mission but when I trigged it via the console it seems to be working so far.

Will removing the mod cause issues with the game?

**Reply by SomeRandomNoobKekeke** [reply]
I think no, but feel free to backup your saves
![barodev small](https://github.com/FakeFishGames/Barotrauma/assets/122838333/726b5291-d904-468b-9d10-9526d5f4f3c8)

**Reply by Regalis11** [reply]
It shouldn't break anything - at worst I'd imagine you might get some errors about some mission not being found if the mod has unlocked missions in a level (which can't be found after the mod has been disabled). But to be safe, it could be a good idea to make a backup of the save, in case the mod was doing something unexpected.
